# 라이브러리 로드하기/데이터 불러오기

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("data/p20v26_KMP_csv.csv")

D:\JadenKim\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (77,86,88,92,93,94,109,133,138,139,140,175,186,199,213,215,216,248,257,280,281,282,290,291,292,348,349,353,355,360) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
df.head()

pid  hid  housenum     p20wt    p20wt2  p20wt_org  p20wt_org2  KMPS10  \
0  10001  100         1  0.136186  0.183348   0.274958    1.907604     NaN   
1  10002  100         2  0.129019  0.161046   0.224794    1.258894     NaN   
2  10003  100         3  0.166530       NaN   0.424265         NaN     NaN   
3  10004  100         4  0.189377       NaN   0.482883         NaN     NaN   
4  20001  200         1  1.582599  1.826189   0.189836         NaN     NaN   

   KMPS11  KMPS19  ...  p20i01034  p20i01018  p20i01019  p20i01030  p20i01020  \
0     1.0     NaN  ...        NaN        2.0        NaN        NaN        NaN   
1     1.0     NaN  ...        NaN        2.0        NaN        NaN        NaN   
2     1.0     NaN  ...        4.0        1.0        1.0        1.0        6.0   
3     1.0     NaN  ...        NaN        NaN        NaN        NaN        NaN   
4     1.0     NaN  ...        NaN        2.0        NaN        NaN        NaN   

   p20i01031  p20i01035  p20i02001  p20i02003  p20i02004  
0        NaN        NaN        2.0        2.0        2.0  
1        NaN        NaN        2.0        1.0        1.0  
2       10.0        4.0        1.0        1.0        1.0  
3        NaN        NaN        2.0        1.0        2.0  
4        NaN        NaN        2.0        1.0        1.0  

[5 rows x 480 columns]

In [4]:
df.shape

(10302, 480)

# 결측치 파악

## 성별, 나이 결측치

In [5]:
df[['p20gender', 'p20age']].isnull().sum()

p20gender    0
p20age       0
dtype: int64

## 학력 결측치

In [6]:
df[['p20school', 'p20school14', 'p20school17']].isnull().sum()

p20school          0
p20school14     9677
p20school17    10286
dtype: int64

학과에 대한 결측치가 9677개로, 입력된 데이터는 625개  
-> 데이터가 지나치게 적으므로, 학과 구분에 따른 정보 리터러시의 차이는 주요하게 다루지 않기로 결정

## 뉴미디어 활용수준 결측치

In [7]:
df[['p20d26007']].isnull().sum()

p20d26007    3275
dtype: int64

In [8]:
df.iloc[:, 141:145].isnull().sum()

p20d27001    5313
p20d27002    5313
p20d27003    5313
p20d27004    5313
dtype: int64

sns 문항 관련 제거해야 하는 결측치 : 5313개

## 매체 활용 능력 결측치

In [9]:
df.iloc[:, 293:314].isnull().sum()

p20d25040    0
p20d25041    0
p20d25042    0
p20d25043    0
p20d25044    0
p20d25045    0
p20d25046    0
p20d25047    0
p20d25048    0
p20d25049    0
p20d25050    0
p20d25051    0
p20d25052    0
p20d25053    0
p20d25054    0
p20d25055    0
p20d25056    0
p20d25057    0
p20d25058    0
p20d25059    0
p20d25060    0
dtype: int64

In [10]:
df.iloc[:, 314:340].isnull().sum()

p20d25061    0
p20d25062    0
p20d25063    0
p20d25064    0
p20d25065    0
p20d25066    0
p20d25067    0
p20d25068    0
p20d25069    0
p20d25070    0
p20d25071    0
p20d25072    0
p20d25073    0
p20d25074    0
p20d25075    0
p20d25076    0
p20d25077    0
p20d25078    0
p20d25079    0
p20d25080    0
p20d25081    0
p20d25082    0
p20d25083    0
p20d25084    0
p20d25085    0
p20d25086    0
dtype: int64

## 전자 상거래 활용 수준 결측치

In [11]:
df.iloc[:, 325:327].isnull().sum()

p20d25072    0
p20d25073    0
dtype: int64

In [12]:
df.iloc[:, 301:303].isnull().sum()

p20d25048    0
p20d25049    0
dtype: int64

In [13]:
df.iloc[:, [458, 464, 470, 476]  ].isnull().sum()

p20i01032    7303
p20i01033    5491
p20i01034    9532
p20i01035    9532
dtype: int64

p20i01032 ~ p20i01035 : 홈쇼핑 관련 문항, 결측치가 최대 9532개로, 정상적으로 입력된 데이터는 770개  
-> 결측치가 너무 많으므로, 해당 문항은 제거  
또한 전자상거래 활용 수준의 문항들은 매체 활용 능력 문항에 포함되므로, '매체 활용 능력'이라는 단일한 변수로 회귀 적합

## 커뮤니티/플랫폼 활용수준 결측치

In [14]:
df.iloc[:, 370:381].isnull().sum()   

p20d14001    0
p20d14002    0
p20d14003    0
p20d14004    0
p20d14005    0
p20d14006    0
p20d14007    0
p20d14008    0
p20d14009    0
p20d14010    0
p20d14011    0
dtype: int64

p20d14012~10414 : 2019년 데이터에서 제외됨

## 프라이버시 민감도 결측치

In [15]:
df.iloc[:, 381:389].isnull().sum() 

p20d23001    0
p20d23002    0
p20d23007    0
p20d23003    0
p20d23004    0
p20d23005    0
p20d23006    0
p20d23008    0
dtype: int64

In [16]:
df.iloc[:, 414:423].isnull().sum() 

p20n01001    406
p20n01002    406
p20n01003    406
p20n01004    406
p20n01005    406
p20n01006    406
p20n01007    406
p20n01008    406
p20n01009    406
dtype: int64

민감도 관련 문항 결측치 406개

# 제거해야 하는 결측치의 총 개수

In [17]:
df[(df['p20d26007'].isnull()==True) | (df['p20d27001'].isnull()==True) | 
   (df['p20d27002'].isnull()==True) | (df['p20d27003'].isnull()==True) | 
   (df['p20d27004'].isnull()==True) | (df['p20n01001'].isnull()==True)].shape

(5774, 480)

SNS 관련 문항이 결측치 or 민감도 관련 문항이 결측치인 데이터의 개수 : 5774개  
-> 분석에 사용할 총 데이터 개수 : 4528개

# p20d26001에서 제거해야 하는 문항 수

In [18]:
df[(df['p20d26001']==8)]['p20d26001'].shape

(0,)

p20d26001 문항에서 8은 인터넷을 사용하지 않는다는 응답  
-> 다행히 8로 응답한 데이터는 존재하지 않음. 추가적인 처리 필요 없음